engine/access/ingestion/engine_test.go

```diff
@@ -112,12 +112,22 @@ func (suite *Suite) SetupTest() {
 		blocksToMarkExecuted, rpcEng)
 	require.NoError(suite.T(), err)
 
-	// stops requestMissingCollections from executing in processBackground worker
-	suite.blocks.On("GetLastFullBlockHeight").Once().Return(uint64(0), errors.New("do nothing"))
+	getLastFullBlockHeightCalls := 0
+	returnErr := func() error {
+		// stops requestMissingCollections from executing in processBackground worker during component startup
+		if getLastFullBlockHeightCalls < 2 {
+			return fmt.Errorf("do nothing")
+		}
```

Comment:
why do we need this now?

Reply:
So that we return an error for the first call to [getLastFullBlockHeight](https://github.com/onflow/flow-go/blob/443e8038838bf488a951d249bf05fb4be4708144/engine/access/ingestion/engine.go#L591) which allows us to [skip requesting collections in the background](https://github.com/onflow/flow-go/blob/443e8038838bf488a951d249bf05fb4be4708144/engine/access/ingestion/engine.go#L218-L239). For all calls after the first call we want to instead return nil now that we are checking the last full block height [here](https://github.com/onflow/flow-go/blob/443e8038838bf488a951d249bf05fb4be4708144/engine/access/ingestion/engine.go#L411).